Dockerfile

```diff
@@ -0,0 +1,37 @@
+# Build stage
+FROM python:3.9-slim AS build
+
+# Set environment variables
+ENV PYTHONUNBUFFERED=1
+
+# Set work directory
+WORKDIR /app
+
+# Copy requirements and install dependencies
+COPY requirements.txt .
+RUN pip install --no-cache-dir -r requirements.txt
+
+# Copy project files
+COPY . .
+
+# Run stage
+FROM python:3.9-slim AS run
+
+# Set environment variables
+ENV PYTHONUNBUFFERED=1
+
+# Set work directory
+WORKDIR /app
+
+# Copy dependencies from build stage
+COPY --from=build /usr/local/lib/python3.9/site-packages /usr/local/lib/python3.9/site-packages
+COPY --from=build /app .
+
+# Execute migrations
+RUN python manage.py migrate

```

Comment:
Running migrations during the image build process might not be ideal, especially if the database is not available at build time. Consider running migrations as part of the container startup process instead.